Components/Payload/Providers/ShopperInfoProvider.php

```diff
@@ -26,7 +26,7 @@ public function provide(PaymentContext $context): array
                 'gender' => $context->getOrder()->getCustomer()->getSalutation(),
             ],
             'shopperLocale' => Shopware()->Shop()->getLocale()->getLocale(),
-            'shopperReference' => $context->getOrder()->getCustomer()->getId(),
+            'shopperReference' => $context->getOrder()->getCustomer()->getNumber(),
```

Comment:
@bortefi so what would be the difference between the customer ID and the Number? Just out of curiosity

Reply:
The **id** (INT 11) is the primary key of customer(=user) table, this could be too short (less than three characters) 
therefore the stored Payment Methods API would not work.

The **number** (VARCHAR 30) is also auto generated in Shopware but can also be manually altered in the Administration.
The default generation is more than 3 characters (internal the table s_order_number is used for incrementing numbers).

_Possible improvement would be that a warning or error is generated when the customer number is too short._